src/Components/ModalWithForm/ModalWithForm.tsx

```diff
@@ -24,116 +35,140 @@ export function ModalWithForm({ movie, onReset, onSubmit }:
     );
   };
 
-  const handleFormChange = (event: ChangeEvent<HTMLElement>) => {
-    const element = event.target as HTMLInputElement;
-
-    if (element.name === 'date') {
-      element.value ? element.classList.add(css.full) : element.classList.remove(css.full);
-    }
-
-    setFormData({
-      ...formData,
-      [element.name]: element.name === 'runtime' ? +element.value : element.value,
-    });
-  };
-
-  const handleSelectChange = (options: { label: string; value: string }[]) => {
-    setFormData({
-      ...formData,
-      genres: options.map(option => option.value),
-    });
-  };
-
-  const formSubmit = () => {
-    onSubmit(formData);
+  const formSubmit = (values) => {
+    onSubmit({ ...movie, ...values });
   };
 
   return (
     <>
       <h2>{formType.toUpperCase()} MOVIE</h2>
-      <form onSubmit={formSubmit}>
-        {formType === 'edit' && (
-          <fieldset>
-            <label>MOVIE ID</label>
-            {movie.id}
-          </fieldset>
+      <Formik
+        initialValues={movie || {}}
+        validationSchema={formValidationSchema}
+        onSubmit={formSubmit}
+      >
+        {({ values, errors, touched, isValid, dirty }) => (
+          <Form>
+            {formType === 'edit' && (
+              <fieldset>
+                <label>MOVIE ID</label>
+                {movie.id}
+              </fieldset>
+            )}
+            <fieldset>
+              <label htmlFor="title">TITLE</label>
+              <Field
+                name="title"
+                placeholder="Movie title"
+                className={errors.title && touched.title && css.notValid}
```

Comment:
It might be better to extract all similar logic into reusable functions to increase readability